drivers/dai/intel/ssp/ssp.c

```diff
@@ -2133,20 +2187,25 @@ static int dai_ssp_set_config_blob(struct dai_intel_ssp *dp, const struct dai_co
 
 	if (blob15->version == SSP_BLOB_VER_1_5) {
 		err = dai_ssp_parse_aux_data(dp, spec_config);
-		if (err) {
+		if (err)
 			return err;
-		}
-		dai_ssp_set_reg_config(dp, cfg, &blob15->i2s_ssp_config);
+		dai_ssp_set_reg_config(dp, cfg, (void *)&blob15->i2s_ssp_config);
 		err = dai_ssp_set_clock_control_ver_1_5(dp, &blob15->i2s_mclk_control);
-		if (err) {
+		if (err)
+			return err;
+	} else if (blob30->version == SSP_BLOB_VER_3_0) {
+		err = dai_ssp_parse_aux_data(dp, spec_config);
+		if (err)
+			return err;
+		dai_ssp_set_reg_config(dp, cfg, (void *)&blob30->i2s_ssp_config);
+		err = dai_ssp_set_clock_control_ver_1_5(dp, &blob30->i2s_mclk_control);
+		if (err)
 			return err;
-		}
 	} else {
-		dai_ssp_set_reg_config(dp, cfg, &blob->i2s_driver_config.i2s_config);
+		dai_ssp_set_reg_config(dp, cfg, (void *)&blob->i2s_driver_config.i2s_config);
 		err = dai_ssp_set_clock_control_ver_1(dp, &blob->i2s_driver_config.mclk_config);
-		if (err) {
+		if (err)
 			return err;
-		}
```

Comment:
You always need the braces in Zephyr, even if you have a single instruction in a branch.